src/redux/books/booksSlice.js

```diff
@@ -1,40 +1,64 @@
-import { createSlice } from '@reduxjs/toolkit';
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+
+const url = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/XbS86wRRQyh8wlRxSdM0/books';
+
+export const getBook = createAsyncThunk('Books/getBook', async () => {
+  const res = await fetch(url);
+  const data = await res.json();
+  const books = [
+    Object.keys(data).map((key) => ({
+      iten_id: key,
+      ...data[key][0],
+    })),
+  ];
+  return books;
+});
+
+export const addBook = createAsyncThunk('Book/addBook', async (Book) => {
+  await fetch(`${url}`, {
+    method: 'POST',
+    body: JSON.stringify(Book),
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+  return Book;
+});
+
+export const removeBook = createAsyncThunk('removeBook', async (id) => {
+  await fetch(`${url}/${id}`, {
+    method: 'DELETE',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+  return id;
+});
```

Comment:
- Great job for setuping redux :+1: however you are dispatching the remove action with an undefined parameter, so your delete functionality is not working both on the API method and redux-store filter :+1: please fix it to have a fully functioning bookstore :+1:

Reply:
please @binhussen  i don't understand